### PR TITLE
Using constructor to create an operator.

### LINKED ISSUE #3438: Make op_registry.{h,cc} call new constructor of OperatorBase.
This is part of https://github.com/PaddlePaddle/Paddle/issues/3425

### PR DESCRIPTION
**Pull request by reyoung**
Fix #3438. The commit is 11c3560.
This PR is based on https://github.com/wangkuiyi/Paddle/pull/13. Because in https://github.com/wangkuiyi/Paddle/pull/13, we changed the private members of OperatorBase. Please review https://github.com/wangkuiyi/Paddle/pull/13 first.

### CODE REVIEW

File: paddle/framework/operator.h [hunk 1]
```diff
@@ -107,53 +102,71 @@ class OperatorBase {
...
+        ret_val.insert(ret_val.end(), out->second.begin(), out->second.end());
+      }
+    }
+    return ret_val;
+  }
 
-  const std::string Type() const { return type_; }
-  const std::vector<std::string> Inputs() const { return inputs_; }
-  const std::vector<std::string> Outputs() const { return outputs_; }
+  std::string Type() const { return type_; }
```
Comment (jacquesqiao) [inline paddle/framework/operator.h]: 这里也应该是 
```cpp
const std::string& Type() const { return type_; }
```

File: paddle/operators/fill_zeros_like_op.cc [hunk 2]
```diff
@@ -18,19 +18,13 @@ namespace paddle {
...
-                      "Input size of FillZerosLikeOp must be one.");
-    PADDLE_ENFORCE_EQ(ctx.OutputSize(), 1UL,
-                      "Output size of AddOp must be one.");
-    PADDLE_ENFORCE_NOT_NULL(ctx.InputVar(0),
-                            "Input of FillZerosLikeOp must be set.");
-    PADDLE_ENFORCE_NOT_NULL(ctx.OutputVar(0),
-                            "Output of FillZerosLikeOp must be set.");
-    ctx.Output<framework::Tensor>(0)->Resize(
-        ctx.Input<framework::Tensor>(0)->dims());
+    ctx.Output<framework::Tensor>("Dst")->Resize(
```
Comment (jacquesqiao) [inline paddle/operators/fill_zeros_like_op.cc]: do not need to check here？

File: paddle/operators/net_op.cc [hunk 3]
```diff
@@ -15,48 +15,42 @@
...
-    if (Contains(temp_output, outputs_[i])) {
-      tmp_index.push_back(i);
+      for (auto& var_name : opt.second) {
+        output_set.insert(var_name);
+      }
     }
   }
-
-  attrs_["temporary_index"] = tmp_index;
+  auto& inputs = inputs_[kAll];
```
Comment (jacquesqiao) [inline paddle/operators/net_op.cc]: kALL对应的是map拍平之后的vector么

File: paddle/operators/fill_zeros_like_op.cc (hunk 2 above)
Comment (reyoung) [inline paddle/operators/fill_zeros_like_op.cc]: 这里因为Ouput("")里做了check了。

File: paddle/framework/operator.h (hunk 1 above)
Comment (reyoung) [inline paddle/framework/operator.h]: 可以用另一个PR修改这里。

**Pull request merged**